toshmanuel/co-sign
# Generateaddress endpoint

**What does this PR do?**
This PR contains the generate address endpoint and get address info endpoint

A summary of what this PR is about

**List of tasks completed**

-   [x] Specify what tasks are completed in this PR
- - users can generate address when they submit their keys
Users can get the address and its redeem script if they want to import the address to an outside wallet.
- Create an endpoint that gets the list of all addresses owned by a user and the balance
- endpoint to generate new address for users to avoid address reuse

**How can this be tested?**
- activate virtual environment
- install all dependencies "pip install -r requirements.txt"
- run "python manage.py runserver",
- For the generate address endpoint:
    - go to "http://127.0.0.1:8000/api/v1/generateaddress"    
    - input the data needed 
    - {
       "key1":first pub key needed,
       "key2":second pubkey,
        "user_id": [in the case of frontend,this can be gotten from the request header I think], make sure you have a registered user
       }

- Highlight how this PR can be tested

**Additional info**

Add more info if any. eg fixes #.
